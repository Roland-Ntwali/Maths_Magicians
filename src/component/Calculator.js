import React from 'react';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.cal = [];
  }

  render() {
    return (
      <div className="calculus">
        <div className="display">
          <span>0</span>
        </div>
        <div className="All">
          <div className="row-1">
            <button type="button">AC</button>
            <button type="button">+/-</button>
            <button type="button">%</button>
            <button type="button" className="orange"> รท</button>
          </div>
          <div className="row-2">
            <button type="button">7</button>
            <button type="button">8</button>
            <button type="button">9</button>
            <button type="button" className="orange">x</button>
          </div>
          <div className="row-3">
            <button type="button">4</button>
            <button type="button">5</button>
            <button type="button">6</button>
            <button type="button" className="orange">-</button>
          </div>
          <div className="row-4">
            <button type="button">1</button>
            <button type="button">2</button>
            <button type="button">3</button>
            <button type="button" className="orange">+</button>
          </div>
          <div className="point">
            <button type="button" className="zero">0</button>
            <button type="button" className="dot">.</button>
            <button type="button" className="orange">=</button>
          </div>
        </div>
      </div>
    );
  }
}
export default Calculator;
