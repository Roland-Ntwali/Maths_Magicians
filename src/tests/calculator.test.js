import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Calculator from '../components/calculator';

it('renders without crashing', () => {
  const page = render(<Calculator />);
  expect(page).toMatchSnapshot();
});

it('it renders simple operation ', () => {
  expect((1 - 1)).toBe(0);
  expect((3 + 4)).toBe(7);
});
