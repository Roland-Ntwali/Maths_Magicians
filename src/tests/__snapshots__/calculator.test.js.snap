// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="calculus"
      >
        <div
          class="display"
        >
          0
        </div>
        <div
          class="All"
        >
          <div
            class="row-1"
          >
            <button
              type="button"
            >
              C
            </button>
            <button
              type="button"
            >
              +/-
            </button>
            <button
              type="button"
            >
              %
            </button>
            <button
              class="orange"
              type="button"
            >
               รท
            </button>
          </div>
          <div
            class="row-2"
          >
            <button
              type="button"
            >
              7
            </button>
            <button
              type="button"
            >
              8
            </button>
            <button
              type="button"
            >
              9
            </button>
            <button
              class="orange"
              type="button"
            >
              x
            </button>
          </div>
          <div
            class="row-3"
          >
            <button
              type="button"
            >
              4
            </button>
            <button
              type="button"
            >
              5
            </button>
            <button
              type="button"
            >
              6
            </button>
            <button
              class="orange"
              type="button"
            >
              -
            </button>
          </div>
          <div
            class="row-4"
          >
            <button
              type="button"
            >
              1
            </button>
            <button
              type="button"
            >
              2
            </button>
            <button
              type="button"
            >
              3
            </button>
            <button
              class="orange"
              type="button"
            >
              +
            </button>
          </div>
          <div
            class="point"
          >
            <button
              class="zero"
              type="button"
            >
              0
            </button>
            <button
              class="dot"
              type="button"
            >
              .
            </button>
            <button
              class="orange"
              type="button"
            >
              =
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="calculus"
    >
      <div
        class="display"
      >
        0
      </div>
      <div
        class="All"
      >
        <div
          class="row-1"
        >
          <button
            type="button"
          >
            C
          </button>
          <button
            type="button"
          >
            +/-
          </button>
          <button
            type="button"
          >
            %
          </button>
          <button
            class="orange"
            type="button"
          >
             รท
          </button>
        </div>
        <div
          class="row-2"
        >
          <button
            type="button"
          >
            7
          </button>
          <button
            type="button"
          >
            8
          </button>
          <button
            type="button"
          >
            9
          </button>
          <button
            class="orange"
            type="button"
          >
            x
          </button>
        </div>
        <div
          class="row-3"
        >
          <button
            type="button"
          >
            4
          </button>
          <button
            type="button"
          >
            5
          </button>
          <button
            type="button"
          >
            6
          </button>
          <button
            class="orange"
            type="button"
          >
            -
          </button>
        </div>
        <div
          class="row-4"
        >
          <button
            type="button"
          >
            1
          </button>
          <button
            type="button"
          >
            2
          </button>
          <button
            type="button"
          >
            3
          </button>
          <button
            class="orange"
            type="button"
          >
            +
          </button>
        </div>
        <div
          class="point"
        >
          <button
            class="zero"
            type="button"
          >
            0
          </button>
          <button
            class="dot"
            type="button"
          >
            .
          </button>
          <button
            class="orange"
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
